src/server-sse.ts

```diff
@@ -0,0 +1,439 @@
+// Constants
+// ============================================================================
+
+const DEFAULT_PORT = 9191;
+const BEARER_PREFIX = "Bearer ";
+
+const HEADERS = {
+	AUTHORIZATION: "authorization",
+	X_SHORTCUT_API_TOKEN: "x-shortcut-api-token",
+	MCP_SESSION_ID: "mcp-session-id",
+	LAST_EVENT_ID: "last-event-id",
+} as const;
+
+const JSON_RPC_ERRORS = {
+	UNAUTHORIZED: { code: -32000, message: "Unauthorized" },
+	BAD_REQUEST: { code: -32000, message: "Bad Request" },
+	SESSION_NOT_FOUND: { code: -32001, message: "Session not found" },
+	INTERNAL_ERROR: { code: -32603, message: "Internal server error" },
+} as const;
+
+// ============================================================================
+// Configuration
+// ============================================================================
+
+interface ServerConfig {
+	port: number;
+	isReadonly: boolean;
+	enabledTools: string[];
+}
+
+function loadConfig(): ServerConfig {
+	let isReadonly = process.env.SHORTCUT_READONLY !== "false";
+	let enabledTools = parseToolsList(process.env.SHORTCUT_TOOLS || "");
+
+	// Parse command line arguments
+	if (process.argv.length >= 3) {
+		process.argv
+			.slice(2)
+			.map((arg) => arg.split("="))
+			.forEach(([name, value]) => {
+				if (name === "SHORTCUT_READONLY") isReadonly = value !== "false";
+				if (name === "SHORTCUT_TOOLS") enabledTools = parseToolsList(value);
+			});
+	}
+
+	return {
+		port: Number.parseInt(process.env.PORT || String(DEFAULT_PORT), 10),
+		isReadonly,
+		enabledTools,
+	};
+}
+
+function parseToolsList(toolsStr: string): string[] {
+	return toolsStr
+		.split(",")
+		.map((tool) => tool.trim())
+		.filter(Boolean);
+}
+
+// ============================================================================
+// Session Management
+// ============================================================================
+
+class SessionManager {
+	private transports: Map<string, StreamableHTTPServerTransport> = new Map();
+
+	has(sessionId: string): boolean {
+		return this.transports.has(sessionId);
+	}
+
+	get(sessionId: string): StreamableHTTPServerTransport | undefined {
+		return this.transports.get(sessionId);
+	}
+
+	add(sessionId: string, transport: StreamableHTTPServerTransport): void {
+		this.transports.set(sessionId, transport);
+		console.log(`Session initialized with ID: ${sessionId}`);
+	}
+
+	remove(sessionId: string): void {
+		this.transports.delete(sessionId);
+		console.log(`Transport closed for session ${sessionId}`);
+	}
+
+	async closeAll(): Promise<void> {
+		console.log("\nShutting down server...");
+		for (const [sessionId, transport] of this.transports.entries()) {
+			try {
+				console.log(`Closing transport for session ${sessionId}`);
+				await transport.close();
+				this.remove(sessionId);
+			} catch (error) {
+				console.error(`Error closing transport for session ${sessionId}:`, error);
+			}
+		}
+		console.log("Server shutdown complete");
```

Comment:
Not sure if it matters or is preferred, but if there are many sessions, they'd be closed one by one serially right now. You could close them all in parallel instead if you did:


```suggestion
		console.log("\nShutting down server...");
		const promises = [];
		for (const [sessionId, transport] of this.transports.entries()) {
				console.log(`Closing transport for session ${sessionId}`);
				promises.push(transport.close().catch(error => {
				  console.error(`Error closing transport for session ${sessionId}:`, error);
				});
				this.remove(sessionId);
		}
		await Promise.all(promises);
		console.log("Server shutdown complete");
```